### PR TITLE
feat: account switching

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

user need to be able to switch between multiple authenticated accounts

### Solutions

updating the ConnectionPolicyManager to observe the current session and update the sync policy whenever the current session is changed

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-906/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-906/3/) **failed**.

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-906/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4283591)(4283591)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4283591)(4283591)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/userprofile/self/SelfUserProfileViewModel.kt [hunk 1]
```diff
@@ -55,7 +61,9 @@ class SelfUserProfileViewModel @Inject constructor(
     private val authServerConfigProvider: AuthServerConfigProvider,
     private val selfServerLinks: SelfServerConfigUseCase,
     private val kaliumConfigs: KaliumConfigs,
-    private val otherAccountMapper: OtherAccountMapper
+    private val otherAccountMapper: OtherAccountMapper,
+    private val updateCurrentSession: UpdateCurrentSessionUseCase,
+    private val observerEstablishedCall: ObserveEstablishedCallsUseCase
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/userprofile/self/SelfUserProfileViewModel.kt:66]:
```suggestion
    private val observeEstablishedCalls: ObserveEstablishedCallsUseCase
```

File: app/src/main/kotlin/com/wire/android/ui/userprofile/self/SelfUserProfileViewModel.kt [hunk 2]
```diff
@@ -64,6 +72,15 @@ class SelfUserProfileViewModel @Inject constructor(
     init {
         viewModelScope.launch {
             fetchSelfUser()
+            observeONGoingCalles()
+        }
+    }
+
+    private fun observeONGoingCalles() {
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/userprofile/self/SelfUserProfileViewModel.kt:79]:
```suggestion
    private fun observeEstablishedCall() {
```

**Review by alexandreferris — APPROVED** [review]
Great work! 

I just have some comments about naming :)

File: app/src/main/kotlin/com/wire/android/ui/userprofile/self/SelfUserProfileScreen.kt [hunk 3]
```diff
@@ -164,7 +169,22 @@ private fun SelfUserProfileContent(
                         }
                         items(
                             items = otherAccounts,
-                            itemContent = { account -> OtherAccountItem(account) }
+                            itemContent = { account ->
+                                OtherAccountItem(
+                                    account,
+                                    clickable = Clickable(enabled = true, onClick = {
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/userprofile/self/SelfUserProfileScreen.kt:175]: I think it's good to wrap `Clickable` with `remember`

File: app/src/main/kotlin/com/wire/android/ui/userprofile/self/SelfUserProfileViewModel.kt [hunk 4]
```diff
@@ -217,6 +234,20 @@ class SelfUserProfileViewModel @Inject constructor(
         }
     }
 
+    fun onOtherAccountClick(userId: UserId) {
+        viewModelScope.launch {
+            when (updateCurrentSession(userId)) {
+                is UpdateCurrentSessionUseCase.Result.Failure -> return@launch
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/userprofile/self/SelfUserProfileViewModel.kt:240]: Shouldn't we handle the failure somehow by at least showing snackbar? 🤔

File: app/src/main/kotlin/com/wire/android/ui/userprofile/self/SelfUserProfileScreen.kt [hunk 5]
```diff
@@ -393,3 +413,10 @@ private fun SelfUserProfileScreenPreview() {
 private fun CurrentSelfUserStatusPreview() {
     CurrentSelfUserStatus(UserAvailabilityStatus.AVAILABLE, onStatusClicked = {})
 }
+
+
+fun Context.getActivity(): AppCompatActivity? = when (this) {
```
[the remark this answers is not in the document]
Comment (MohamadJaara) [reply app/src/main/kotlin/com/wire/android/ui/userprofile/self/SelfUserProfileScreen.kt:418]: Ohhh my bad I was testing something and forgot to remove it

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-906/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4288259)(4288259)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4288259)(4288259)-dev-release.apk)

**Pull request merged**